### DISCUSSION TITLE
How to drop key-value pairs in catn in malli?

### DISCUSSION THREAD

**Discussion by glyh** [metosin/malli#801] [Q&A] [answered]
I'm writing a parser for my language, here's what I got:
```clj
(def ifl
  [:schema
   {:registry
    {::defn ;:int
     [:catn [:_ [:= 'defn]] [:name :symbol] [:bindings [:vector :symbol]] [:exp :any]]
     ::defns
     [:sequential [:ref ::defn]]}}
   ::defns])
```
note on line 5 I was forced to use a label `:_` to store the value I don't want, how should I drop this kv pair from the map? 

I realize that under some circumstances, it will make it impossible to unparse, so probably that is impossible to achieve?

**Comment by ikitommi** [comment, marked as answer]
There is no built-in way to drop the unwanted keys. You could `clojure.walk/prewalk` them away, but as you said, you can't unparse it if you drop them. I just would not drop them.
